src/video/wayland/SDL_waylandwindow.c

```diff
@@ -812,6 +813,50 @@ void Wayland_HideWindow(_THIS, SDL_Window *window)
     }
 }
 
+static void
+handle_xdg_activation_done(void *data,
+                           struct xdg_activation_token_v1 *xdg_activation_token_v1,
+                           const char *token)
+{
+    SDL_WindowData *window = data;
+    SDL_assert(xdg_activation_token_v1 == window->activation_token);
+    xdg_activation_token_v1_destroy(window->activation_token);
+    window->activation_token = NULL;
+}
+
+static const struct xdg_activation_token_v1_listener activation_listener_xdg = {
+    handle_xdg_activation_done
+};
+
+void Wayland_RaiseWindow(_THIS, SDL_Window *window)
+{
+    SDL_VideoData *data = _this->driverdata;
+    SDL_WindowData *wind = window->driverdata;
+    const char *app_name;
+
+    if (data->activation_manager) {
+        if (wind->activation_token != NULL) {
+            /* We're about to overwrite this with a new request */
+            xdg_activation_token_v1_destroy(wind->activation_token);
+        }
+
+        wind->activation_token = xdg_activation_v1_get_activation_token(data->activation_manager);
+        xdg_activation_token_v1_add_listener(wind->activation_token,
+                                             &activation_listener_xdg,
+                                             wind);
+
+        /* FIXME: Maybe remove the "audio" part of this hint's name? -flibit */
+        app_name = SDL_GetHint(SDL_HINT_AUDIO_DEVICE_APP_NAME);
+        if (!app_name || *app_name == '\0') {
+            app_name = "SDL Application";
+        }
+
+        xdg_activation_token_v1_set_surface(wind->activation_token, wind->surface);
```

Comment:
If possible, call `xdg_activation_token_v1_set_serial` with the seat+serial of the event (e.g. click or key press) that triggered the activation. Some compositors might only raise windows if they do this.

Reply:
Added a note regarding the serial - this call is technically arbitrary and not based on reactions to any events, so we won't have a serial readily available. That said, it's okay if this gets mostly ignored, especially since it's arbitrary.